app/helpers/mock-resources.ts

```diff
@@ -114,3 +118,36 @@ export function createSamplePrivateCloudCommentData(args?: { data?: Partial<Pris
 
   return _data;
 }
+
+export function getUserServiceAccountAuthHeader(user?: AppUserWithRoles) {
+  const stringifiedRoles = _join(user?.roles, ',');
+  const signature = jws.sign({
+    header: { alg: 'HS256', typ: 'JWT' },
+    payload: {
+      roles: stringifiedRoles,
+      service_account_type: 'user',
+      'kc-userid': user?.id,
+    },
+    secret,
+  });
+  return {
+    Authorization: 'Bearer ' + signature,
+    'Content-Type': 'application/json',
+  };
+}
+
+export function getTeamServiceAccountAuthHeader() {
+  const signature = jws.sign({
+    header: { alg: 'HS256', typ: 'JWT' },
+    payload: {
+      roles: 'private-admin,public-admin',
```

Comment:
when convert the method more generic, then the logic should support it as well.. does all service account have the same roles?
- also please create a function `getServiceAccountAuthHeader` that encapsulates the common logic from both `getUserServiceAccountAuthHeader` and `getTeamServiceAccountAuthHeader`